server/channels/app/channel.go

```diff
@@ -863,12 +878,24 @@ func (a *App) PatchChannel(c request.CTX, channel *model.Channel, patch *model.C
 	oldChannelHeader := channel.Header
 	oldChannelPurpose := channel.Purpose
 
+	if *a.Config().ExperimentalSettings.ExperimentalChannelCategorySorting && strings.Contains(*patch.DisplayName, "/") {
+		parts := strings.Split(*patch.DisplayName, "/")
+		*patch.DisplayName = strings.Join(parts[1:], "/")
+		channel.DefaultCategoryName = strings.TrimSpace(parts[0])
+	}
+
 	channel.Patch(patch)
 	channel, err := a.UpdateChannel(c, channel)
 	if err != nil {
 		return nil, err
 	}
 
+	user, err := a.GetUser(userID)
+	if err != nil {
+		return nil, err
+	}
+	a.addChannelToDefaultCategory(c, user, channel)
```

Comment:
We are only using the userID in this method. So unless it's needed, I'd recommend to simply pass userID and avoid a `a.GetUser` call.